exporters/trace_stackdriver/src/main/java/io/opencensus/trace/exporter/StackdriverV1ExporterHandler.java

```diff
@@ -0,0 +1,164 @@
+import com.google.devtools.cloudtrace.v1.Traces;
+import io.grpc.Channel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.auth.MoreCallCredentials;
+import io.opencensus.common.Function;
+import io.opencensus.common.Functions;
+import io.opencensus.common.Timestamp;
+import io.opencensus.trace.AttributeValue;
+import io.opencensus.trace.SpanContext;
+import io.opencensus.trace.SpanId;
+import io.opencensus.trace.TraceId;
+import io.opencensus.trace.export.SpanData;
+import io.opencensus.trace.export.SpanExporter.Handler;
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * A class that contains helper methods to convert {@link SpanData} to Stackdriver Trace API v1
+ * trace messages.
+ */
+final class StackdriverV1ExporterHandler extends Handler {
+  private static final String API_HOST = "cloudtrace.googleapis.com";
+  private static final String STATUS_CODE = "g.co/status/code";
+  private static final String STATUS_DESCRIPTION = "g.co/status/description";
+
+  private final String projectId;
+  private final TraceServiceBlockingStub traceServiceBlockingStub;
+
+  StackdriverV1ExporterHandler(Credentials credentials, String projectId) {
+    checkNotNull(credentials, "credentials");
+    this.projectId = checkNotNull(projectId, "projectId");
+    Channel channel = ManagedChannelBuilder.forTarget(API_HOST).build();
+    traceServiceBlockingStub =
+        TraceServiceGrpc.newBlockingStub(channel)
+            .withCallCredentials(MoreCallCredentials.from(credentials));
+  }
+
+  Trace generateTrace(SpanData spanData) {
+    SpanContext context = spanData.getContext();
+    TraceSpan.Builder spanBuilder =
+        TraceSpan.newBuilder()
+            .setSpanId(toSpanIdProto(context.getSpanId()))
+            .setKind(toSpanKindProto(spanData))
+            .setName(spanData.getName())
+            .setStartTime(toTimestampProto(spanData.getStartTimestamp()))
+            .setEndTime(toTimestampProto(spanData.getEndTimestamp()));
+
+    if (spanData.getParentSpanId() != null && spanData.getParentSpanId().isValid()) {
+      spanBuilder.setParentSpanId(toSpanIdProto(spanData.getParentSpanId()));
+    }
+
+    for (Map.Entry<String, AttributeValue> label :
+        spanData.getAttributes().getAttributeMap().entrySet()) {
+      spanBuilder.putLabels(label.getKey(), attributeValueToString(label.getValue()));
+    }
+
+    // Add Status as labels in the v1 API.
+    spanBuilder.putLabels(STATUS_CODE, spanData.getStatus().getCanonicalCode().toString());
+    if (spanData.getStatus().getDescription() != null) {
+      spanBuilder.putLabels(STATUS_DESCRIPTION, spanData.getStatus().getDescription());
+    }
+
+    Trace.Builder traceBuilder =
+        Trace.newBuilder()
+            .setProjectId(projectId)
+            .setTraceId(toTraceIdProto(context.getTraceId()))
+            .addSpans(spanBuilder.build());
+
+    return traceBuilder.build();
+  }
+
+  private static long toSpanIdProto(SpanId spanId) {
```

Comment:
Why do `toSpanIdProto` and `toTraceIdProto` have "proto" in the name?

Reply:
Is the encoding that is used in the proto request. Maybe other suggestion?

Reply:
Maybe `encodeSpanId` or `toTraceSpanId`?